Update mapping. pkg to release 1.4.3

PRN183 opened the pull request:
A new mapping pkg release.
Would one of the devs here please merge?
Thanks

mmuetzel commented:
The CI failed. That was probably because of an issue with the CI rules that should be fixed now.
Could you please rebase your PR on a current head of the main branch?

mmuetzel commented:
Never mind: Closing and re-opening the PR re-runs the PR on a (rebased) merge branch.

mmuetzel commented on packages/mapping.yaml:
```diff
@@ -24,6 +24,15 @@ maintainers:
+  - "octave (>= 5.2.0)"
+  - "io (>= 2.2.7)"
+  - "geometry (>= 4.0.0)"
+  - "pkg"
```
``` suggestion
  - "pkg"
 ubuntu2204:
  - "libgdal-dev"
```

That would indicate to the CI to install the GDAL package (for building and running with more enabled features).

mmuetzel commented on packages/mapping.yaml:
```diff
@@ -24,6 +24,17 @@ maintainers:
+  - "io (>= 2.2.7)"
+  - "geometry (>= 4.0.0)"
+  - "pkg"
+ ubuntu2204:
```
``` suggestion
  ubuntu2204:
```

Oops. I messed up indentation.

siko1056 commented:
@mmuetzel Thank you for looking into this.

I think the current latest problem is that if the list of versions is not 100% identical, jsondecode makes a Cell array rather than an array of struct.  This bug came with the latest switch to JSON.

Are you working on this now?

@PRN183 Apologies for the delay.

mmuetzel commented:
> I think the current latest problem is that if the list of versions is not 100% identical, jsondecode makes a Cell array rather than an array of struct. This bug came with the latest switch to JSON.

I came to the same conclusion.

> Are you working on this now?

See #573.

mmuetzel commented:
I missed a couple more places. 😖

Will take a couple hours before I can come back to this.

PRN183 commented:
I saw libgdal added as dependency, but it is merely a suggested one. The mapping package builds fine w/o gdal but then some functionality isn't available ("XFAIL").

@siko1056 : "@PRN183 Apologies for the delay."
I never saw such fast response to a PR or submitted patch; and I didn't notice there has been quite some frantic activity while I went outside for some personal things.

@mmuetzel : thanks very much for fixing the details and the merge!

mmuetzel commented:
> I saw libgdal added as dependency, but it is merely a suggested one. The mapping package builds fine w/o gdal but then some functionality isn't available ("XFAIL").

The `ubuntu2204` dependencies are (mostly) just for the CI run here. I.e., the tests that would otherwise be "ignored" (by XFAIL) should actually work here.

mmuetzel commented:
> @mmuetzel : thanks very much for fixing the details and the merge!

I didn't merge. (In fact, I would have waited a bit for your reaction regarding the documentation issue.)

PRN183 commented:
@mmuetzel 
_"The mapping.txi file contains @include functions.texi (pretty close to the end of the file). But afaics, there is no functions.texi file in that directory in the tarball."_

Didn't know that functions.texi, or for that matter, all .texi files are needed.
AFAIU esp. functions.texi is a mere intermediate file, just like mapping.html, only used to create the GUI Qt docs files mapping.qhc and mapping.qch by some .py scripts. I removed it by hand from a copy of the mapping package repo before running 'make release'.
I think that next time all the .texi + .py + .css files + maybe a few more had better be dropped from the doc/ subdir before creating a tarball as they aren't needed for running the package; maybe just let mapping.info live on, plus of course the .qch and .qhc files. I left the .texi in thinking that it's easier for users if they want to fix some functionality or documentation in an installed package. That way users don't need to clone the repo for just a maybe simple fix. Just running 'make doc' in the doc/ subdir will re-generate functions.texi.
FWIW, I compared with the signal and ?mttq? (forgot name) packages to see how 'lostbard' managed to get the functions docs in the GUI; but in the releases he made of various packages there are is no consistent file contents.
FYI the GUI docs Makefile I included in <mapping>/doc/ is a cleaned up one I copied over from the signal package.  In the io package it didn't work anymore to run 'make release' so I just did it my way for the time being and moved it to doc/.

Or has the "general" package Makefile changed for all packages, so that Qt GUI docs are generated? The package repos on SF still have a Makefile from some years ago.

BTW:  I made this mapping pkg release similar to that of the io package yesterday. (AFAIK) I also removed functions.texi from the io-2.7.0 tarball.

PRN183 commented:
BTW I'll be away very soon, don't know if and how much time I'll have this weekend for Octave stuff.

mmuetzel commented:
Thanks for clarifying.

If I understand correctly, no changes are actually needed for the tarball. I don't think many users with ever run `doctest` for this package. So, we can just ignore these failing doctests.

How to proceed with this (stop including these .texi files in the tarball or keep them in) is up to you. I just figured I'd ask because I noticed those messages in the CI output here.

PRN183 commented:
Actually I have no idea what happens once I add a PR here. Apparently there are a lot of checks on the packages (good!). But I didn't find any description yet of what the required/suggested setup for a package is these days, and exactly which package contents are checked.

AFAIK doc/ used to be a non-tracked subdir, as data/ and others. So I didn't know that the CI (what is that anyway?) would dive into it and check even package .texi files that AFAIK are nowhere used within Octave.

All I can say is that I checked that the package "works" in the sense that it can be installed (and thus 'built'), loaded, the functions work, no FAILs are seen with 'pkg test', and the GUI doc files are registered when the package gets loaded. What else would need to be checked as far as the tarball goes?

siko1056 commented:
@mmuetzel This CI mostly automates the checks from Octave Forge https://wiki.octave.org/Reviewing_Octave_Forge_packages, that is `pkg test` and `doctest` only.

In most packages `doctest` is not used, given the delicate nature of relying on strings and output format settings by the user.

What the package does, please be kindly referred to https://gnu-octave.github.io/packages/doctest/ :innocent:

PRN183 commented:
@siko1056 
No worries, I think there's a misunderstanding between us, due to a gross misunderstanding by me about how packages.octave.org works these days. No new release is required.

Yes it took me a while to get used to the new workflow. No problem, that's life.

Point is, I didn't expect automated package tests, maybe only some tests for .yaml syntax and if the .yaml entries point to the right places.
I did all the tests set out in "reviewing packages" for the mapping package myself, except for doctest. So I was a bit surprised to learn that "CI" (whatever that is) double-checked quite a bit more, so to say.  (BTW I did doctest for the io package and AFAICS it didn't add much useful stuff, so I skipped it for the mapping package.)
A thing I expect that the CI doesn't do is checking if the package installs and runs in the oldest Octave release "supported by" the package. I still have Octave-4.2.0 installed for that purpose w.r.t. io package, plus one of each major release.

Anyway, overall I am quite pleased with how fast a package release goes, compared to the old SF release system.

W.r.t. your other message about rsync/docs (can't find the thread now) (io.html and mapping.html): in the end I used filezilla, that's much easier than rsync. I checked all file permissions in SF, filezilla makes that quite easy to do. I did check if it all works "GNU Octave- packages" and AFAICS it does :-)

OK what's left for me is to await bug reports so that a new release gets on the horizon some time :-)

mmuetzel commented:
> Point is, I didn't expect automated package tests, maybe only some tests for .yaml syntax and if the .yaml entries point to the right places.

I don't think that there is a hard requirement that a PR must pass all checks here before it can be merged. But in some (maybe many?) cases, a failing test might indicate that something is wrong.
Unfortunately, these automated tests can give wrong results at times. (E.g., for some packages the files in the `doc` directory might matter. For others, they might not.) So, manual review is still needed.
The tests that are run here are probably not sufficient. Package maintainers should still run their own tests before a release.
However, I still think that running these tests here is better than not running them. I hope that asking about some aspects of the results of these automated tests isn't too much of a bother to package maintainers.

> "CI" (whatever that is)

https://en.wikipedia.org/wiki/Continuous_integration

PRN183 commented:
> So, manual review is still needed.
My impression was that the packages.octave thing was set up to relieve the admins from that sort of burden ...

Stupid question: where can I see the results from the CI runners when I upload the packages and enter a PR? or do they only run after the merge?

> However, I still think that running these tests here is better than not running them.
O yes, agreed.

BTW I'll upload a patch for mxe-octave within a few minutes to the patch tracker.

mmuetzel commented:
> Stupid question: where can I see the results from the CI runners when I upload the packages and enter a PR? or do they only run after the merge?

There are no stupid questions. 🙂
You can access the results on the "Checks" tab that you can find on top of the original comment of the PR.
They used to be accessible also in a box just above the box where you write new comments. But apparently, that box disappears now at some point. (Maybe once a PR is merged?)

I'm not sure why GitHub is indicating that 6 checks have been run for this PR. Usually, it is just two checks: One for valid yaml syntax, and another one that tries to install the package in Octave (usually in the latest released version) and tries to run some tests with it.
For a PR which displays just the usual two checks, see, e.g., #564.
